sym_api_client_python/services/abstract_datafeed_event_service.py

```diff
@@ -0,0 +1,322 @@
+from abc import ABC, abstractmethod
+import logging
+import json
+import time
+
+from ..listeners.elements_listener import ElementsActionListener
+from ..listeners.connection_listener import ConnectionListener
+from ..listeners.im_listener import IMListener
+from ..listeners.room_listener import RoomListener
+
+from ..auth.auth_endpoint_constants import auth_endpoint_constants
+
+log = logging.getLogger(__name__)
+
+class AbstractDatafeedEventService(ABC):
+
+    def __init__(self, sym_bot_client, log_events=True, error_timeout_sec=None, maximum_timeout_sec=None):
+        self.datafeed_events = []
+        self.room_listeners = []
+        self.im_listeners = []
+        self.connection_listeners = []
+        self.elements_listeners = []
+        self.suppression_listeners = []
+        self.wall_post_listeners = []
+        self.registered_triggers = []
+
+        self.bot_client = sym_bot_client
+        self.datafeed_client = self.bot_client.get_datafeed_client()
+        self.log_events = log_events
+        self.stop = False
+
+        config = sym_bot_client.get_sym_config()
+        # TODO: Should not be handled in the DF ES like this, why put a timeout in the config if we can put a parameter with default value(config related)
+        # Timeout will start at and eventually reset to this
+        _config_key = 'datafeedEventsErrorTimeout'
+        # Mapping of Event type to the function to handle it
+        self.routing_dict = {
+            'MESSAGESENT': self.msg_sent_handler,
+            'MESSAGESUPPRESSED': self.suppressed_message_handler,
+            'INSTANTMESSAGECREATED': self.instant_msg_handler,
+            'ROOMCREATED': self.room_created_handler,
+            'ROOMDEACTIVATED': self.room_deactivated_handler,
+            'ROOMREACTIVATED': self.room_reactivated_handler,
+            'ROOMUPDATED': self.room_updated_handler,
+            'USERJOINEDROOM': self.user_joined_room_handler,
+            'USERLEFTROOM': self.user_left_room_handler,
+            'ROOMMEMBERPROMOTEDTOOWNER': self.promoted_to_owner,
+            'ROOMMEMBERDEMOTEDFROMOWNER': self.demoted_to_owner,
```

Comment:
This is a detail but a big ambiguous, it should be `self.demoted_from_owner` if we want to stay consistent.

Reply:
of course.